ddtrace/tracer/textmap.go

```diff
@@ -114,74 +115,188 @@ func NewPropagator(cfg *PropagatorConfig) Propagator {
+// context. By default, only the datadog propagation style will be used.
+// If the DD_PROPAGATION_STYLE_INJECT environment variable is set,
+// this can override the default behavior.
+func makeInjectors(cfg *PropagatorConfig) []textMapPropagator {
+	dd := &datadogPropagator{cfg}
+	b3 := &b3Propagator{}
+
+	ps := os.Getenv("DD_PROPAGATION_STYLE_INJECT")
+	if ps == "" {
+		return []textMapPropagator{dd}
+	}
+	styles := propagationStyleList(ps)
+
+	var injectors []textMapPropagator
+	for _, v := range styles {
+		switch v {
+		case "Datadog":
+			injectors = append(injectors, dd)
+		case "B3":
+			injectors = append(injectors, b3)
+		default:
+			// TODO: consider logging something for invalid/unknown styles.
+		}
+	}
+
+	// If all the styles were invalid/unknown, then revert to default behavior.
+	if len(injectors) == 0 {
+		return []textMapPropagator{dd}
+	}
+	return injectors
+}
+
+func makeExtractors(cfg *PropagatorConfig) []textMapPropagator {
+	dd := &datadogPropagator{cfg}
+	b3 := &b3Propagator{}
+
+	ps := os.Getenv("DD_PROPAGATION_STYLE_EXTRACT")
+	if ps == "" {
+		return []textMapPropagator{dd}
+	}
+	styles := propagationStyleList(ps)
+
+	var extractors []textMapPropagator
+	for _, v := range styles {
+		switch v {
+		case "Datadog":
+			extractors = append(extractors, dd)
+		case "B3":
+			extractors = append(extractors, b3)
+		default:
+			// TODO: consider logging something for invalid/unknown styles.
+		}
+	}
+
+	// If all the styles were invalid/unknown, then revert to default behavior.
+	if len(extractors) == 0 {
+		return []textMapPropagator{dd}
+	}
+	return extractors
+}
+
+// propagationStyleList splits a string containing propagation styles
+// separated by space or comma, and returns a []string containing
+// only the propagation styles.
+func propagationStyleList(s string) []string {
+	f := func(r rune) bool {
+		switch r {
+		case ' ', ',':
+			return true
+		}
+		return false
+	}
+	return strings.FieldsFunc(s, f)
+}
 
 // Inject defines the Propagator to propagate SpanContext data
 // out of the current process. The implementation propagates the
 // TraceID and the current active SpanID, as well as the Span baggage.
 func (p *propagator) Inject(spanCtx ddtrace.SpanContext, carrier interface{}) error {
-	switch v := carrier.(type) {
+	switch c := carrier.(type) {
 	case TextMapWriter:
-		return p.injectTextMap(spanCtx, v)
+		// Apply all of the injectors.
+		for _, v := range p.injectors {
+			err := v.injectTextMap(spanCtx, c)
+			if err != nil {
+				return err
+			}
+		}
+		return nil
 	default:
 		return ErrInvalidCarrier
 	}
 }
 
-func (p *propagator) injectTextMap(spanCtx ddtrace.SpanContext, writer TextMapWriter) error {
+// Extract implements Propagator.
+func (p *propagator) Extract(carrier interface{}) (ddtrace.SpanContext, error) {
+	switch c := carrier.(type) {
+	case TextMapReader:
+		// Try each extractor. The first to successfully produce a context
```

Comment:
Can we do the same here with this comment? We could just put it inside the `if` statement. I feel it belongs there:
```go
if ctx != nil {
    // first successful extractor returns
    return ctx, nil
}
```